apps/comps/hooks/useAgent.ts

```diff
@@ -19,21 +19,16 @@ export const useUserAgent = (id?: string): UseQueryResult<Agent, Error> => {
 };
 
 /**
- * Hook to fetch a single agent by ID (unauthenticated)
+ * Hook to fetch a single agent by ID (public, unauthenticated)
  * @param id Agent ID
  * @returns Query result with agent data
  */
-export const useAgent = (id?: string) => {
-  return useQuery({
-    queryKey: ["agent", id],
-    queryFn: async (): Promise<AgentWithOwnerResponse> => {
-      if (!id) throw new Error("Agent ID is required");
-      const response = await apiClient.getAgent(id);
-
-      if (!response.success) throw new Error("Error when querying agent");
-
-      return response;
-    },
-    enabled: !!id,
-  });
-};
+export const useAgent = (
+  id?: string,
+): UseQueryResult<RouterOutputs["agent"]["getAgent"], Error> =>
+  useQuery(
+    tanstackClient.agent.getAgent.queryOptions({
+      input: { agentId: id! },
+      enabled: !!id,
```

Comment:
I learned the hard way this doesn't work. The query options are evaluated immediately, so `input` is calculated right away with no consideration of the `enabled` property, so `id!` will throw an exception. You just need to set `agentId` conditionally like 
```
input: { agentId: id || "" },
```